File: testcases/src/main/java/recursive/inheritance_and_recursive/ICRUD.java
package recursive.inheritance_and_recursive;

import javax.ws.rs.DELETE;
import javax.ws.rs.POST;
import javax.ws.rs.Path;
import javax.ws.rs.PathParam;

public interface ICRUD {

    @POST
    @Path("create")
    public String createREST(String transferObject);


    @DELETE
    @Path("delete/{id}")
    public String deleteREST(@PathParam("id") String id);
}

File: testcases_kt/src/main/kotlin/recursive/inheritance_and_recursive/ICRUD.kt
package recursive.inheritance_and_recursive

import javax.ws.rs.DELETE
import javax.ws.rs.POST
import javax.ws.rs.Path
import javax.ws.rs.PathParam

interface ICRUD {

    @DELETE
    @Path("delete/{id}")
    fun deleteREST(@PathParam("id") id: String): String
}
